drivers/spi/spi_sam.c

```diff
@@ -128,7 +128,11 @@ static int spi_sam_configure(const struct device *dev,
 	 * select mode.
 	 */
 	spi_mr |= (SPI_MR_MSTR | SPI_MR_MODFDIS);
-	spi_mr |= SPI_MR_PCS(spi_slave_to_mr_pcs(config->slave));
+	if (spi_cs_is_gpio(config)) {
+		spi_mr |= SPI_MR_PCS(spi_slave_to_mr_pcs(0));
+	} else {
+		spi_mr |= SPI_MR_PCS(spi_slave_to_mr_pcs(config->slave));
+	}
```

Comment:
I still fail to understand why it wouldn't work without this change, but in any case you can shorten the statement:

```suggestion
	spi_mr |= SPI_MR_PCS(spi_slave_to_mr_pcs(spi_cs_is_gpio(config) ? 0 : config->slave));
```

Same for the `CSR`.